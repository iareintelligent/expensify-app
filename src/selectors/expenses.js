import moment from "moment";

export default (expenses, { text, sortBy, startDate, endDate }) => {
    return expenses
        .filter(expense => {
            const createdAtMoment = moment(expense.createdAt);
            const startDateMatch = startDate
                ? startDate.isSameOrBefore(createdAtMoment, "day")
                : true;
            const endDateMatch = endDate
                ? endDate.isSameOrAfter(createdAtMoment, "day")
                : true;
            // const textMatch = expense.toLowerCase().includes(text) ? text : false;
            const textMatch = expense.description.toLowerCase().includes(text);

            return startDateMatch && endDateMatch && textMatch;
        })
        .sort((a, b) => {
            switch (sortBy) {
                case "amount":
                    return b.amount - a.amount;
                case "date":
                    return a.createdAt - b.createdAt;
            }
        });
};
