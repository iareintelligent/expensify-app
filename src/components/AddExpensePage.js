import React from "react";
import ExpenseForm from "./ExpenseForm";
import { addExpense } from "../actions/expenses";
import { connect } from "react-redux";

export class AddExpensePage extends React.Component {
    onSubmit = expense => {
        // props.dispatch(addExpense(expense));
        this.props.addExpense(expense);
        this.props.history.push("/");
    };
    render() {
        return (
            <div>
                <h1>Add Expense</h1>
                <ExpenseForm onSubmit={this.onSubmit} />
            </div>
        );
    }
}

// const mapStateToProps = state => {
//     return {
//         expenses: state.expenses
//     };
// };

const mapDispatchToProps = dispatch => ({
    addExpense: expense => dispatch(addExpense(expense))
});

// export default connect(mapStateToProps)(AddExpensePage);
export default connect(
    undefined,
    mapDispatchToProps
)(AddExpensePage);
