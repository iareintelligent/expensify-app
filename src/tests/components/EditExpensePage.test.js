import React from "react";
import { shallow } from "enzyme";
import { EditExpensePage } from "../../components/EditExpensePage";
import expenses from "../fixtures/expenses";

let editExpense, removeExpense, history, wrapper;
const expense = expenses[2];

beforeEach(() => {
    editExpense = jest.fn();
    removeExpense = jest.fn();
    history = { push: jest.fn() };

    wrapper = shallow(
        <EditExpensePage
            editExpense={editExpense}
            removeExpense={removeExpense}
            history={history}
            expense={expenses[2]}
        />
    );
});

test("should render the EditExpensePage correctly", () => {
    expect(wrapper).toMatchSnapshot();
});

test("should handle editExpense", () => {
    wrapper.find("ExpenseForm").prop("onSubmit")(expense);
    expect(editExpense).toHaveBeenLastCalledWith(expense.id, expense);
    expect(history.push).toHaveBeenLastCalledWith("/");
});

test("should handle removeExpense", () => {
    wrapper.find('button').simulate('click');
    expect(removeExpense).toHaveBeenLastCalledWith({ id: expense.id });
    expect(history.push).toHaveBeenLastCalledWith("/");
});
