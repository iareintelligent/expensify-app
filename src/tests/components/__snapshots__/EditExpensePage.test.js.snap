// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the EditExpensePage correctly 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 1000,
        "createdAt": 345600000,
        "description": "Coffee",
        "id": 3,
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove Expense
  </button>
</div>
`;
